src/org/thoughtcrime/securesms/NewConversationActivity.java

```diff
@@ -87,6 +95,29 @@ private void handleIntent() {
     }
   }
 
+  private String getTextToShare(Uri uri) {
+    Map<String, String> mailtoQueryMap = getMailtoQueryMap(uri);
+    String textToShare = mailtoQueryMap.get(SUBJECT);
+    String body = mailtoQueryMap.get(BODY);
+    if (body != null && !body.isEmpty()) {
+      textToShare += "\n" + body;
+    }
+    return textToShare;
+  }
+
+  private Map<String, String> getMailtoQueryMap(Uri uri) {
```

Comment:
Manual query "parsing" and "handling" as the MailTo stuff given by the systems fails ...